src/cli/self_update.rs

```diff
@@ -65,7 +65,7 @@ pub const NEVER_SELF_UPDATE: bool = false;
 // argument of format! needs to be a literal.
 
 macro_rules! pre_install_msg_template {
-    ($platform_msg: expr) => {
+    ($platform_msg:literal) => {
```

Comment:
This commit seems unrelated to your other work, why did you do it?

Reply:
The macro argument of that macro is always literal to me.
I'd move it to another PR (this change is so small) if you prefer.